es/es-repository-s3/src/test/java/org/elasticsearch/repositories/s3/S3ClientSettingsTests.java

```diff
@@ -0,0 +1,123 @@
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package org.elasticsearch.repositories.s3;
+
+import com.amazonaws.ClientConfiguration;
+import com.amazonaws.Protocol;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.auth.BasicSessionCredentials;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.test.ESTestCase;
+import org.junit.Test;
+
+import java.util.Map;
+
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.isEmptyString;
+import static org.hamcrest.Matchers.nullValue;
+
+public class S3ClientSettingsTests extends ESTestCase {
+
+    @Test
+    public void testThereIsADefaultClientByDefault() {
+        final Map<String, S3ClientSettings> settings = S3ClientSettings.load(Settings.EMPTY);
+        assertThat(settings.keySet(), contains("default"));
+
+        final S3ClientSettings defaultSettings = settings.get("default");
+        assertThat(defaultSettings.credentials, nullValue());
+        assertThat(defaultSettings.endpoint, isEmptyString());
+        assertThat(defaultSettings.protocol, is(Protocol.HTTPS));
+        assertThat(defaultSettings.proxyHost, isEmptyString());
+        assertThat(defaultSettings.proxyPort, is(80));
+        assertThat(defaultSettings.proxyUsername, isEmptyString());
+        assertThat(defaultSettings.proxyPassword, isEmptyString());
+        assertThat(defaultSettings.readTimeoutMillis, is(ClientConfiguration.DEFAULT_SOCKET_TIMEOUT));
+        assertThat(defaultSettings.maxRetries, is(ClientConfiguration.DEFAULT_RETRY_POLICY.getMaxErrorRetry()));
+        assertThat(defaultSettings.throttleRetries, is(ClientConfiguration.DEFAULT_THROTTLE_RETRIES));
+    }
+
+    @Test
+    public void testDefaultClientSettingsCanBeSet() {
+        final Map<String, S3ClientSettings> settings = S3ClientSettings.load(Settings.builder()
+                                                                                 .put("s3.client.default.max_retries", 10)
+                                                                                 .build());
+        assertThat(settings.keySet(), contains("default"));
+
+        final S3ClientSettings defaultSettings = settings.get("default");
+        assertThat(defaultSettings.maxRetries, is(10));
+    }
+
+    @Test
+    public void testRejectionOfLoneAccessKey() {
+        final IllegalArgumentException e = expectThrows(IllegalArgumentException.class,
+                                                        () -> S3ClientSettings.load(Settings.builder()
+                                                                                        .put("s3.client.default.access_key", "aws_key")
+                                                                                        .build()));
```

Comment:
we use the `ExpectedException` Rule to expect exception, could we stick to that style instead (also everywhere in this suite)?